catalog/spatial/registry/registry-api-impl/src/main/java/org/codice/ddf/registry/api/impl/RegistryStoreImpl.java

```diff
@@ -136,6 +161,31 @@ public SourceResponse query(QueryRequest request) throws UnsupportedQueryExcepti
             return new SourceResponseImpl(request, Collections.emptyList());
         }
 
+        SourceResponse registryQueryResponse = queryResponse(request);
+        for (Result singleResult : registryQueryResponse.getResults()) {
+            if (singleResult.getMetacard()
+                    .getAttribute(RegistryObjectMetacardType.REGISTRY_ID)
+                    .getValue()
+                    .toString()
+                    .equals(registryId)) {
```

Comment:
how many of these do you expect to match this id? you could probably use some Java 8 streaming stuff, like .find (or findFirst) to grab the one that matches and then do the updating if necessary

you maybe could even inline that as well, with a .map() call?